core/src/crds_gossip_push.rs

```diff
@@ -102,67 +105,58 @@ impl CrdsGossipPush {
     ) -> Vec<Pubkey> {
         let origin_stake = stakes.get(origin).unwrap_or(&0);
         let self_stake = stakes.get(self_pubkey).unwrap_or(&0);
-        let cache = self.received_cache.get(origin);
-        if cache.is_none() {
-            return Vec::new();
-        }
-        let peers = cache.unwrap();
-
+        let peers = match self.received_cache.get_mut(origin) {
```

Comment:
I always have a bit of. trouble remembering what is being tracked in the `received_cache`

iirc, the `received_cache` is more like a `active_push_validators_by_origin`, where for each possible origin `X`, it tells us which set of validators `V`  have pushed messages `M` to us where `M.origin == X`. Could we add a quick comment to `received_cache` to clarify this? Thanks 😃!

Reply:
Agree that it is confusing; It has some comments here:
https://github.com/solana-labs/solana/blob/0aa782488/core/src/crds_gossip_push.rs#L52-L56
I also added comments for what the fields are as in above.